atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -86,11 +89,31 @@ private static ThreadFactory createThreadFactory(String safeLoggablePurpose) {
     public ListenableFuture<R> apply(T argument) {
         Preconditions.checkState(!closed, "Autobatcher is already shut down");
         DisruptorFuture<R> result = new DisruptorFuture<R>(safeLoggablePurpose);
-        buffer.publishEvent((refresh, sequence) -> {
+        buffer.publishEvent(createTranslator(argument, result));
+        return result;
+    }
+
+    public List<ListenableFuture<R>> applyBatch(List<T> arguments) {
+        Preconditions.checkState(!closed, "Autobatcher is already shut down");
+        List<ListenableFuture<R>> results = new ArrayList<>();
+
+        EventTranslator<DefaultBatchElement<T, R>>[] translators = arguments.stream().map(argument -> {
+            DisruptorFuture<R> result = new DisruptorFuture<>(safeLoggablePurpose);
+            EventTranslator<DefaultBatchElement<T, R>> translator = createTranslator(argument, result);
+            results.add(result);
+            return translator;
+        }).toArray(EventTranslator[]::new);
+
+        buffer.publishEvents(translators);
+
+        return results;
```

Comment:
Tests for this method, and I suppose for this whole class? The batching code is tricky, so definitely need tests for batching code.